### PR TITLE
Added the user sign up/in features as well as models

### PR DESCRIPTION
**Pull request by code-matt**
added Devise user authentication
added FactoryGirl
Added the user sign up/in feature tests and made them pass.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,2 +1,92 @@
...
+  config.mock_with :rspec do |mocks|
+    # Prevents you from mocking or stubbing a method that does not exist on
+    # a real object. This is generally recommended, and will default to
+    # `true` in RSpec 4.
+    mocks.verify_partial_doubles = true
+  end
+
+# The settings below are suggested to provide a good initial experience
+# with RSpec, but feel free to customize to your heart's content.
+=begin
```
Comment (smkopp92) [inline spec/spec_helper.rb:45]: Address this. I'll use the :confused: emoji for hound issues you should fix.

File: spec/features/users/user_signs_up_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  scenario 'Provided signup information is invalid' do
+    visit root_path
+    click_link 'Sign Up'
+    click_button 'Sign Up'
+
+    expect(page).to have_content("can't be blank")
+    expect(page).to_not have_content("Sign Out")
+  end
+
+
```
Comment (mlg-) [inline spec/features/users/user_signs_up_spec.rb:29]: Fix this

File: db/migrate/20160405152355_devise_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+class DeviseCreateUsers < ActiveRecord::Migration
+  def change
+    create_table(:users) do |t|
+      ## Database authenticatable
+      t.string :first_name, null: false
+      t.string :last_name, null: false
+
```
Comment (mlg-) [inline db/migrate/20160405152355_devise_create_users.rb:7]: Fix this

File: db/migrate/20160405152355_devise_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+
```
Comment (mlg-) [inline db/migrate/20160405152355_devise_create_users.rb:36]: Fix this

File: spec/factories.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :user do
+    sequence(:email) {|n| "person#{n}@example.com" }
```
Comment (mlg-) [inline spec/factories.rb:3]: Fix this

File: config/routes.rb [hunk 6]
```diff
@@ -1,10 +1,12 @@
...
+  # devise_for :users
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
   # You can have the root of your site routed with "root"
-  # root 'welcome#index'
-
+  resources :users, only: [:index]
+  root 'users#index'
+  devise_for :users, :controllers => { registrations: 'registrations' }
```
Comment (mlg-) [inline config/routes.rb:9]: Fix this

**Pull request merged**